app/sweepers.py

```diff
@@ -0,0 +1,76 @@
+
+"""
+This module contains the definitions of Sagittariidae's sweeper processes.
+These are asynchronous processes that designed to drive the current state of
+the world to a desired final state.
+
+The sweepers are run as processes separate from the webservice.  They may be
+invoked by name by executing this module, typically from a scheduler such as
+`cron`.
+"""
+
+import os
+import shutil
+import sys
+
+import app as sagittariidae
+import models
+
+
+logger = sagittariidae.app.logger
+
+class Sweeper(object):
+
+    def run(self):
+        raise NotImplementedError()
+
+    def sweep(self):
+        try:
+            self.run()
+        except Exception, e:
+            logger.error('Unhandled exception in sweeper %s', self, exc_info=e)
```

Comment:
Yes, it is.  Not allowing the exception to bubble up means that we don't run the risk of the sweeper getting stuck on a broken file.  It'll simply log the error and move on, doing as much work as it can on each run.

To make sure that we don't lose sight of errors in the sweepers and in the service, we'll need something that will aggregate these error reports and send us alarms when they reach unreasonable levels.